zerver/views/auth.py

```diff
@@ -296,6 +296,7 @@ def register_remote_user(request: HttpRequest, result: ExternalAuthResult) -> Ht
     kwargs.pop("subdomain", None)
     kwargs.pop("redirect_to", None)
     kwargs.pop("is_realm_creation", None)
+    kwargs.pop("params_to_store_in_authenticated_session", None)
```

Comment:
Can you explain the rule for which things should be popped here? I realize it's things that `maybe_send_to_registration` doesn't accept as a parameter ... but I don't understand which things need to be plumbed through to that function, i.e. why these 4 parameters are the things that belong here.

Reply:
These keys are listed in `ExternalAuthDataDict` (that's the data type of `result.data_dict`). I pushed a commit to add a comment mention of this:
```
    # maybe_send_to_registration doesn't take these arguments, so delete them.
    # The full list of keys carried in this dict can be found in ExternalAuthDataDict
    # definition.
    kwargs.pop("subdomain", None)
    kwargs.pop("redirect_to", None)
    kwargs.pop("is_realm_creation", None)
    kwargs.pop("params_to_store_in_authenticated_session", None)
```

Reply:
I see the amended comment is later in the series.

But it seems like this might be better as an explicit include-list, rather than popping off some set of things it _can't_ handle?  That way we reduce action-at-a-distance when more keys get added into `ExternalAuthDataDict` -- and we instead have an explicit include list of kwargs that `maybe_send_to_registration` takes, right above the call to `maybe_send_to_registration`.

We could even use [`inspect.signature`](https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object) to build the include-list automatically -- but I think having to be explicit about what we're plumbing to the function call, from the data structure pulled from deep in this code, is probably a feature.